src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -769,6 +771,252 @@ protected bool IsAlive
+                    break;
+            }
+            return components;
+        }
+
+        private static List<Element> GetChildElementsFromFamilyInstance(Autodesk.Revit.DB.Element element)
+        {
+            var familyInstance = element as Autodesk.Revit.DB.FamilyInstance;
+            if (familyInstance == null)
+                throw new NullReferenceException(Properties.Resources.NoChildElements);
+            List<ElementId> componentIds = familyInstance.GetSubComponentIds().ToList();
+            if (componentIds.Count == 0 || componentIds == null)
+                throw new NullReferenceException(Properties.Resources.NoChildElements);
+            List<Element> componentElements = componentIds.Select(id => Document.GetElement(id).ToDSType(true))
+                                                           .ToList();
+            return componentElements;
+        }
+
+        private static List<Element> GetChildElementsFromRailings(Autodesk.Revit.DB.Element element)
+        {
+            var railingElement = element as Autodesk.Revit.DB.Architecture.Railing;
+            List<ElementId> railingComponentIds = new List<ElementId>();
+            railingComponentIds.AddRange(railingElement.GetHandRails().ToList());
+            railingComponentIds.Add(railingElement.TopRail);
+            if (railingComponentIds.Count == 0)
+                throw new InvalidOperationException(Properties.Resources.NoChildElements);
+            List<Element> railingComponentElements = railingComponentIds.Select(id => Document.GetElement(id).ToDSType(true))
+                                                                         .ToList();
+            return railingComponentElements;
+        }
+
+        private static List<Element> GetChildElementsFromStructuralFramingSystem(Autodesk.Revit.DB.Element element)
+        {
+            var beamSystemElement = element as Autodesk.Revit.DB.BeamSystem;
+            List<ElementId> beamSystemComponentIds = beamSystemElement.GetBeamIds().ToList();
+            if (beamSystemComponentIds.Count == 0)
+                throw new InvalidOperationException(Properties.Resources.NoChildElements);
+            List<Element> beamSystemComponentElements = beamSystemComponentIds.Select(id => Document.GetElement(id).ToDSType(true))
+                                                                     .ToList();
+            return beamSystemComponentElements;
+        }
+
+        private static List<Element> GetChildElementsFromStairs(Autodesk.Revit.DB.Element element)
+        {
+            var stairElement = element as Autodesk.Revit.DB.Architecture.Stairs;
+            List<ElementId> stairComponentIds = new List<ElementId>();
+            stairComponentIds.AddRange(stairElement.GetStairsLandings().ToList());
+            stairComponentIds.AddRange(stairElement.GetStairsRuns().ToList());
+            stairComponentIds.AddRange(stairElement.GetStairsSupports().ToList());
```

Comment:
Can we add more comments here indicating why these are counted as child elements or point to an online reference? I feel like there is some hidden knowledge here for a Revit newbie like me

Reply:
agree, @SHKnudsen let's add some notes with the knowledge/info we gathered with Revit team